examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -58,100 +57,376 @@ eprint = {https://doi.org/10.1175/MWR2930.1}
   #Pressure
-  H = Rm_sfc * T_BL / grav;
-  P = P_sfc * exp(-xvert/H);
-  #Exner
-  exner_dry = exner(P, PhasePartition(DT(0)))
-  #Temperature 
-  T             = exner_dry*θ_liq + LH_v0*q_liq/(cpm*exner_dry);
-  #Density
-  ρ             = P/(Rm*T);
-  #Potential Temperature
-  θv     = virtual_pottemp(T, P, q_pt)
-  # energy definitions
-  u, v, w     = DT(7), DT(-5.5), DT(0)
-  U           = ρ * u
-  V           = ρ * v
-  W           = ρ * w
-  e_kin       = DT(1//2) * (u^2 + v^2 + w^2)
+  H     = Rm_sfc * T_sfc / grav;
+  p     = P_sfc * exp(-xvert/H);
+  #Density, Temperature
+  TS    = LiquidIcePotTempSHumEquil_no_ρ(θ_liq, q_pt, p)
+  ρ     = air_density(TS)
+  T     = air_temperature(TS)
+  #Assign State Variables
+  u, v, w     = FT(7), FT(-5.5), FT(0)
+  e_kin       = FT(1/2) * (u^2 + v^2 + w^2)
   e_pot       = grav * xvert
   E           = ρ * total_energy(e_kin, e_pot, T, q_pt)
   state.ρ     = ρ
-  state.ρu    = SVector(U, V, W) 
+  state.ρu    = SVector(ρ*u, ρ*v, ρ*w) 
   state.ρe    = E
   state.moisture.ρq_tot = ρ * q_tot
-end   
+end
+
+
+# TODO: temporary; move to new CLIMA module
+# TODO: add an option to reduce communication: compute averages
+# locally only
+function gather_diagnostics(dg, Q)
+  mpirank = MPI.Comm_rank(MPI.COMM_WORLD)
+  nranks = MPI.Comm_size(MPI.COMM_WORLD)
+
+  grid = dg.grid
+  topology = grid.topology
+  N = polynomialorder(grid)
+  Nq = N + 1
+  Nqk = dimensionality(grid) == 2 ? 1 : Nq
+  nrealelem = length(topology.realelems)
+  nvertelem = topology.stacksize
+  nhorzelem = div(nrealelem, nvertelem)
+
+  nstate = 6
+  nthermo = 5
+  host_array = Array ∈ typeof(Q).parameters
+  localQ = host_array ? Q.realdata : Array(Q.realdata)
+  thermoQ = zeros(Nq * Nq * Nqk, nthermo, nrealelem)
+  vgeo = grid.vgeo
+  localvgeo = host_array ? vgeo : Array(vgeo)
+
+  for e in 1:nrealelem  
+    for i in 1:Nq*Nq*Nqk
+      z = localvgeo[i,grid.x3id,e]
+
+      rho_node  = localQ[i,1,e]
+      u_node    = localQ[i,2,e] / localQ[i,1,e]
+      v_node    = localQ[i,3,e] / localQ[i,1,e]
+      w_node    = localQ[i,4,e] / localQ[i,1,e]
+      etot_node = localQ[i,5,e]/localQ[i,1,e]
+      qt_node   = localQ[i,6,e]/localQ[i,1,e]
+
+      e_int = etot_node - 1//2 * (u_node^2 + v_node^2 + w_node^2) - grav * z
+
+      ts = PhaseEquil(e_int, qt_node, rho_node)
+      Phpart = PhasePartition(ts)
+      thermoQ[i,1,e] = Phpart.liq
+      thermoQ[i,2,e] = Phpart.ice
+      thermoQ[i,3,e] = qt_node-Phpart.liq-Phpart.ice
+      thermoQ[i,4,e] = ts.T
+      thermoQ[i,5,e] = liquid_ice_pottemp(ts)
+    end
+  end
+
+  fluctT = zeros(Nq * Nq * Nqk, nthermo, nrealelem)    
+  fluctQ = zeros(Nq * Nq * Nqk, nstate, nrealelem)
+  varQ   = zeros(Nq * Nq * Nqk, nstate, nrealelem)
+
+  rho_localtot = sum(localQ[:, 1, :])
+  U_localtot   = sum(localQ[:, 2, :])
+  V_localtot   = sum(localQ[:, 3, :])
+  W_localtot   = sum(localQ[:, 4, :])
+  e_localtot   = sum(localQ[:, 5, :])
+  qt_localtot  = sum(localQ[:, 6, :])
```

Comment:
Care with notation: in lines 141 and 164, `qt_<subscript>` refer to different values that vary by the density scaling (the `_<subscript>`s don't contain this information); I'd suggest following the notation in `AtmosModel.jl` for consistency e.g. \rhoq_tot etc.